src/java/org/apache/cassandra/service/accord/serializers/TableMetadatasAndKeys.java

```diff
@@ -0,0 +1,270 @@
+import org.apache.cassandra.schema.TableMetadata;
+import org.apache.cassandra.service.accord.api.PartitionKey;
+import org.apache.cassandra.service.accord.serializers.TableMetadatas.Multi;
+import org.apache.cassandra.utils.btree.BTreeSet;
+
+import static accord.primitives.Routable.Domain.Range;
+import static accord.primitives.Routables.Slice.Minimal;
+
+public class TableMetadatasAndKeys extends IVersionedWithKeysSerializer.AbstractWithKeysSerializer implements Sliceable<TableMetadatasAndKeys>
+{
+    public static class KeyCollector extends AbstractSortedCollector<PartitionKey, Keys>
+    {
+        private static final Comparator<Object> comparator = Comparator.comparing(v -> ((PartitionKey) v));
+
+        public final TableMetadatas tables;
+
+        public KeyCollector(TableMetadatas tables)
+        {
+            this.tables = tables;
+        }
+
+        public TableMetadatasAndKeys buildTablesAndKeys()
+        {
+            return new TableMetadatasAndKeys(tables, build());
+        }
+
+        @Override
+        Comparator<Object> comparator()
+        {
+            return comparator;
+        }
+
+        public PartitionKey collect(TableMetadata table, DecoratedKey key)
+        {
+            TableId tableId = tables.get(table.id).id;
+            if (count == 1)
+            {
+                PartitionKey one = (PartitionKey) buffer;
+                if (one.prefix() == table && one.partitionKey().equals(key))
+                    return one;
+            }
+            return collect(new PartitionKey(tableId, key));
+        }
+
+        @Override
+        Keys empty()
+        {
+            return Keys.EMPTY;
+        }
+
+        @Override
+        Keys of(PartitionKey one)
+        {
+            return Keys.of(one);
+        }
+
+        @Override
+        Keys copy(Object[] array, int count)
+        {
+            Key[] result = new Key[count];
+            System.arraycopy(array, 0, result, 0, count);
+            return Keys.ofSortedUnique(result);
+        }
+
+        @Override
+        Keys copyBtree(Object[] btree, int count)
+        {
+            return Keys.ofSortedUnique(new BTreeSet<>(btree, comparator()));
+        }
+    }
+
+    private static final TableMetadatasAndKeys NO_KEYS = new TableMetadatasAndKeys(Multi.NONE, Keys.EMPTY);
+    private static final TableMetadatasAndKeys NO_RANGES = new TableMetadatasAndKeys(Multi.NONE, Ranges.EMPTY);
+
+    public static TableMetadatasAndKeys none(Routable.Domain domain)
+    {
+        return domain.isKey() ? NO_KEYS : NO_RANGES;
+    }
+
+    public final TableMetadatas tables;
+    public final Seekables keys;
+
+    public TableMetadatasAndKeys(TableMetadatas tables, Seekables keys)
+    {
+        this.tables = tables;
+        this.keys = keys;
+    }
+
+    public TableMetadata get(TableId tableId)
+    {
+        return tables.get(tableId);
+    }
+
+    public void serializeSeekables(Seekables seekables, DataOutputPlus out) throws IOException
```

Comment:
this method is actually dead code?

Reply:
huh, so it is. It wasn't in an earlier version. Ok, removing.